src/cdk-lib/amazonaurora/aurora-default-vector-store.ts

```diff
@@ -0,0 +1,257 @@
+      resourceName: auroraCluster.clusterIdentifier,
+      arnFormat: cdk.ArnFormat.COLON_RESOURCE_NAME,
+    });
+
+    auroraCluster.addRotationSingleUser();
+
+    auroraSecurityGroup.addIngressRule(
+      lambdaSecurityGroup,
+      ec2.Port.tcp(5432),
+      'Allow PostgreSQL access from Lambda security group',
+    );
+
+    // Add Data API access to the Aurora cluster
+    const cfnDbCluster = auroraCluster.node.defaultChild as rds.CfnDBCluster;
+    cfnDbCluster.addOverride('Properties.EnableHttpEndpoint', true);
+
+    this.auroraPgCRPolicy = new iam.ManagedPolicy(this, 'AuroraPgPolicy', {
+      managedPolicyName: generatePhysicalNameV2(this,
+        'AuroraPgPolicy',
+        { maxLength: 32, lower: true }),
+      statements: [
+        new iam.PolicyStatement({
+          actions: [
+            'ec2:DescribeInstances',
+            'ec2:CreateNetworkInterface',
+            'ec2:AttachNetworkInterface',
+            'ec2:DescribeNetworkInterfaces',
+            'autoscaling:CompleteLifecycleAction',
+            'ec2:DeleteNetworkInterface',
+          ],
+          resources: ['*'],
+        }),
+        new iam.PolicyStatement({
+          actions: [
+            'rds-data:BatchExecuteStatement',
+            'rds-data:BeginTransaction',
+            'rds-data:CommitTransaction',
+            'rds-data:ExecuteStatement',
+            'rds-data:RollbackTransaction',
+          ],
+          resources: [
+            cdk.Stack.of(this).formatArn({
+              service: 'rds',
+              resource: 'cluster',
+              resourceName: `${auroraCluster.clusterIdentifier}`,
+            }),
+          ],
+        }),
+      ],
+    });
+
+    NagSuppressions.addResourceSuppressions(
+      this.auroraPgCRPolicy,
+      [
+        {
+          id: 'AwsSolutions-IAM4',
+          reason: 'The AWSLambdaBasicExecutionRole managed policy is required for ' +
+                  'the Lambda function to write logs to CloudWatch.',
+        },
+        {
+          id: 'AwsSolutions-IAM5',
+          reason: 'This policy is required to allow the custom resource to create a ' +
+                  'network interface for the Aurora cluster and it has to be wildcard.',
+        },
+      ],
+      true,
+    );
+
+    NagSuppressions.addResourceSuppressions(
+      auroraCluster,
+      [
+        {
+          id: 'AwsSolutions-RDS10',
+          reason: 'Deletion protection is disabled to make sure a customer can stop ' +
+                  'incurring charges if they want to delete the construct.',
+        },
+      ],
+      true,
+    );
+
+    const customResource = buildCustomResourceProvider({
+      providerName: 'AmazonAuroraPgVectorCRProvider',
+      vpc: vpc,
+      securityGroup: lambdaSecurityGroup,
+      codePath: path.join(
+        __dirname, '../../../lambda/amazon-aurora-pgvector-custom-resources'),
+      handler: 'custom_resources.on_event',
+      runtime: cdk.aws_lambda.Runtime.PYTHON_3_12,
```

Comment:
can we please change the runtime to 3_11 to be in consistent with other constructs.

Reply:
Sure.

Reply:
it should be fine, all the other custom resources in cdk-lib we have use 3.12

Reply:
resolving following comment above